metricbeat/module/redis/redis.go

```diff
@@ -208,27 +208,23 @@ func (p *Pool) DBNumber() int {
 }
 
 // CreatePool creates a redis connection pool
-func CreatePool(
-	host, password, network string,
-	dbNumber int,
-	maxConn int,
-	idleTimeout, connTimeout time.Duration,
-) *Pool {
+func CreatePool(host, username, password string, dbNumber int, config *ModuleConfig, connTimeout time.Duration) *Pool {
```

Comment:
The Dial func could be refactored to accept the `*tls.Config` and append the two additional TLS related options if it's enabled, what do you think?

```suggestion
func CreatePool(host, username, password string, dbNumber int, config *ModuleConfig, connTimeout time.Duration, tlsConfig *tls.Config) *Pool {
```

```go
		Dial: func() (rd.Conn, error) {
			dialOptions := []rd.DialOption{
				rd.DialUsername(username),
				rd.DialPassword(password),
				rd.DialDatabase(dbNumber),
				rd.DialConnectTimeout(connTimeout),
				rd.DialReadTimeout(connTimeout),
				rd.DialWriteTimeout(connTimeout),
			}

			if config.TLS.IsEnabled() {
				dialOptions = append(dialOptions, rd.DialUseTLS(true))
				dialOptions = append(dialOptions, rd.DialTLSConfig(tlsConfig))
			}

			return rd.Dial(config.Network, host, dialOptions...)
		},
```

Reply:
Good point. As I was using an extra field `UseTLS`, I kept it like this. But I think your way is more explicit and would be easier for readers who are trying to understand the code. I'll make this change.

Reply:
Done!